backpack/utils/conv.py

```diff
@@ -15,24 +15,32 @@ def unfold_func(module):
     )
 
 
-def get_conv1d_weight_gradient_factors(input, grad_out, module):
-    # shape [N, C_in * K_x, L_out]
-    X = unfold_by_conv(input, module)
-    return X, grad_out
+def get_weight_gradient_factors(input, grad_out, module, N):
+    # shape [N, C_in * K_x * K_y, H_out * W_out]
+    X = unfold_by_conv(module.input0, module)
 
+    if N == 1:
+        dE_dY = grad_out
+    elif N == 2:
+        dE_dY = eingroup("n,c,h,w->n,c,hw", grad_out)
+    elif N == 3:
+        dE_dY = eingroup("n,c,d,h,w->n,c,dhw", grad_out)
+    else:
+        raise ValueError("{}-dimensional Conv. is not implemented.".format(N))
 
-def get_weight_gradient_factors(input, grad_out, module):
-    # shape [N, C_in * K_x * K_y, H_out * W_out]
-    X = unfold_func(module)(input)
-    dE_dY = eingroup("n,c,h,w->n,c,hw", grad_out)
```

Comment:
For `N==2`, `unfold_func` should be used as it uses `torch.nn.Unfold`, which is faster than our `unfold_by_conv` generalization to 1 and 3 dimensions.